File: ios/Plugin/PlayerPlugin.swift
import Foundation
import Capacitor
import AVKit
import AVFoundation

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(PlayerPlugin)
public class PlayerPlugin: CAPPlugin {
    private let implementation = Player()
    
    @objc func play(_ call: CAPPluginCall) {
        guard let url = call.getString("url") else {
            call.reject("Video URL cannot be empty");
            return
        }
        
        guard let bridge = self.bridge else { return }
        
        let videoURL = URL(string: url)
        
        let player = AVPlayer(url: videoURL!)
        let playerViewController = AVPlayerViewController()
        playerViewController.player = player
        
        DispatchQueue.main.async {
            bridge.viewController?.present(playerViewController, animated: true) {
                playerViewController.player!.play()
                call.resolve()
            }
        }
    }
}

File: android/src/main/java/com/felipemarcos/capacitor/player/PlayerPlugin.java
package com.felipemarcos.capacitor.player;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

import android.content.Intent;
import android.net.Uri;

import java.net.URLConnection;

@CapacitorPlugin(name = "Player")
public class PlayerPlugin extends Plugin {

    private Player implementation = new Player();

    @PluginMethod
    public void play(PluginCall call) {
        String url = call.getString("url");

        if (url == null) {
            call.reject("Video URL cannot be empty");
            return;
        }

        Uri videoUrl = Uri.parse(url);
        String mimeType = URLConnection.guessContentTypeFromName(url);


        Intent intent = new Intent(Intent.ACTION_VIEW, videoUrl);
        intent.setDataAndType(videoUrl, mimeType);
        getActivity().startActivity(intent);

        call.resolve();
    }
}
